src/main/java/ch/njol/skript/ScriptLoader.java

```diff
@@ -1016,9 +986,37 @@ public static ArrayList<TriggerItem> loadItems(SectionNode node) {
 			} else if (subNode instanceof SectionNode) {
 				TypeHints.enterScope(); // Begin conditional type hints
 
-				item = Section.parse(expr, "Can't understand this section: " + expr, (SectionNode) subNode, items);
-				if (item == null)
+				RetainingLogHandler handler = SkriptLogger.startRetainingLog();
+				find_section:
+				try {
+					item = Section.parse(expr, "Can't understand this section: " + expr, (SectionNode) subNode, items);
+					if (item != null)
+						break find_section;
+
+					// back up the failure log
+					RetainingLogHandler backup = handler.backup();
+					handler.clear();
+
+					item = Statement.parse(expr, "Can't understand this effect: " + expr, (SectionNode) subNode, items);
+
+					if (item != null)
+						break find_section;
+					Collection<LogEntry> errors = handler.getErrors();
+
+					// restore the failure log
+					if (errors.isEmpty()) {
+						handler.restore(backup);
+					} else { // We specifically want these two errors in preference to the section error!
+						String firstError = errors.iterator().next().getMessage();
+						if (!firstError.contains("is a valid statement but cannot function as a section (:)")
+							&& !firstError.contains("You cannot have two section-starters in the same line"))
```

Comment:
No. Errors are a terrible mess to handle and this was about the only way we could find to "catch" these specific ones. I think Sovde was looking at improvements to the error system, but it really needs to be a follow-up PR because there isn't an alternative that's in-scope here.